lib/mayaUsd/resources/scripts/mayaUsdAddMayaReference.py

```diff
@@ -49,47 +53,158 @@ def createPrimAndAttributes(stage, primPath, mayaReferencePath, mayaNamespace, m
+
+    The variant set and name are optional
+
+    Parameters:
+    -----------
+    ufePathStr : str : Ufe PathString of parent prim to add Maya Reference
+    mayaReferencePath : str : File path of Maya Reference (for attribute)
+    mayaNamespace : str : Namespace (for attribute)
+    mayaReferencePrimName : str [optional] : Name for the Maya Reference prim
+    groupPrim : tuple(str,str,str) [optional] : The Group prim Name, Type & Kind to create
+                                                Note: the name is optional and will be auto-computed
+                                                      if empty or not provided.
+                                                Note: Type and Kind are both mandatory, but Kind is
+                                                      allowed to be empty string.
+    variantSet : tuple(str,str) [optional] : The Variant Set Name and Variant Name to create
+
+    Return:
+    -------
+    The Usd prim of the newly created Maya Reference or an invalid prim if there is an error.
+    '''
 
     # Make sure the prim name is valid and doesn't already exist.
     parentPrim = mayaUsd.ufe.ufePathToPrim(ufePathStr)
 
-    checkName = mayaUsd.ufe.uniqueChildName(parentPrim, mayaReferencePrimName)
+    # There are special conditions when we are given the ProxyShape gateway node.
+    ufePath = ufe.PathString.path(ufePathStr)
+    isGateway = (ufePath.nbSegments() == 1)
+
+    # Were we given a Group prim to create?
+    groupPrimName = None
+    groupPrimType = None
+    groupPrimKind = None
+    if groupPrim:
+        if (len(groupPrim) == 2):
+            groupPrimType, groupPrimKind = groupPrim
+        elif (len(groupPrim) == 3):
+            groupPrimName, groupPrimType, groupPrimKind = groupPrim
+
+            # Make sure the input Group prim name doesn't exist already
+            # and validate the input name.
+            # Note: it is allowed to be input as empty in which case a default is used.
+            if groupPrimName:
+                checkGroupPrimName = mayaUsd.ufe.uniqueChildName(parentPrim, groupPrimName)
+                if checkGroupPrimName != groupPrimName:
+                    errorMsgFormat = getMayaUsdLibString('kErrorGroupPrimExists')
+                    errorMsg = cmds.format(errorMsgFormat, stringArg=(groupPrimName, ufePathStr))
+                    om.MGlobal.displayError(errorMsg)
+                    return Usd.Prim()
+                groupPrimName = Tf.MakeValidIdentifier(checkGroupPrimName)
+
+        # If the group prim was either not provided or empty we use a default name.
+        if not groupPrimName:
+            groupPrimName = getDefaultGroupPrimName(parentPrim, mayaNamespace)
+
+    # When the input is a gateway we cannot have in variant unless group is also given.
+    if isGateway and variantSet and not groupPrimName:
+        errorMsg = getMayaUsdLibString('kErrorCannotAddToProxyShape')
+        om.MGlobal.displayError(errorMsg)
+        return Usd.Prim()
```

Comment:
Case we discussed about adding Maya Reference directly under ProxyShape. Okay when no variant used. When variant is used, group must also be used.